{id3lib,game-music-emu,mpd,a52dec,yabai,handbrake,cmus}: fix on aarch64-darwin

emilazy opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

The third in the series of quick Apple Silicon fixes, and maybe the last batch for now.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

emilazy committed:
id3lib: add libiconv on Darwin

emilazy committed:
game-music-emu: replace with libgme

These appear to be two independent packages of exactly the same piece of
software! Replace the one that's broken on aarch64-darwin with the one
that isn't.

emilazy committed:
mpd: don't override stdenv on LLVM >= 8

This fixes the build on aarch64-darwin.

emilazy committed:
liba52: 0.7.4 -> unstable-2009-12-08

The SourceForge site hasn't seen a release since 2002; the VideoLAN
repository has over 7 years of additional changes and fixes, including
quite a few by the maintainer listed on the SF site, so I'm going to
take it as the canonical repository. I'm not sure if these ever saw
formal release, so I've just tagged it with a commit date.

VLC needs a trivial patch to accommodate a slight API change; this was
previously reported and rejected in 2010
(https://code.videolan.org/videolan/vlc/-/issues/3731) and a patch was
shared and considered, but never applied, in 2012
(https://mailman.videolan.org/pipermail/vlc-devel/2012-January/084325.html).
I considered backing out this update in light of this, but I think
carrying the tiny VLC patch is probably the better approach;
other downstream consumers like GStreamer already handle both APIs, and
the unreleased commits include some important-looking fixes (including
potential UB).

All of this is probably irrelevant in practice, since VLC prefers using
libavcodec to decode AC3 where possible, and I imagine other software
either follows suit or ought to. Arguably the liba52 package should just
be removed.

emilazy committed:
liba52: don't disable PIC

I'm hoping this fixes the test failure on i686, though I haven't tested
it yet.

emilazy committed:
liba52: rename from a52dec

liba52 is the library; a52dec is the example program it ships with. Most
(all?) of the packages depending on this use it for the library; a52dec
is left as an alias for users.

emilazy committed:
liba52: patch out strange always_inline hack

This fixes the build on aarch64-darwin.

emilazy committed:
yabai: 3.3.4 -> 3.3.10

Contains fixes for aarch64-darwin.

emilazy committed:
handbrake: 1.3.3 -> 1.4.1

Contains fixes for aarch64-darwin.

emilazy committed:
cmus: add VideoToolbox dependency on Darwin

Fixes the build on aarch64-darwin.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 137942` at 9d82782a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>obs-studio-plugins.obs-ndi</li>
    <li>odpdown</li>
    <li>termplay</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/290757ec3f3a6c284c7bc804e01094b4">arcan.arcan</a></li>
    <li><a href="https://gist.github.com/8129f1e0dc72e6979219b98c1941e9b5">arcan.xarcan</a></li>
    <li><a href="https://gist.github.com/d9cbeebfe1ba5fc3c21442bfab8c3685">libvlc</a></li>
    <li><a href="https://gist.github.com/b1eb733fb41142162a6351cc64372f84">pympress</a></li>
    <li><a href="https://gist.github.com/b5844615bd38b48c4e4b812e790d3c42">python38Packages.python-vlc</a></li>
    <li><a href="https://gist.github.com/30df0098cdce540b92aeb18f88a8fdd2">python39Packages.python-vlc</a></li>
    <li><a href="https://gist.github.com/2bdd049324e1c48c4b6d7f73375fe527">reaper</a></li>
    <li><a href="https://gist.github.com/9fa7ff4cc6097e5b8c4340b69fa66004">vlc</a> (vlc_qt5)</li>
    <li><a href="https://gist.github.com/f1ef746055738001661b1b02e89f858b">zdoom</a></li>
  </ul>
</details>
<details>
  <summary>80 packages skipped due to time constraints:</summary>
  <ul>
    <li>MIDIVisualizer</li>
    <li>alarm-clock-applet</li>
    <li>audio-recorder</li>
    <li>beets</li>
    <li>beetsExternalPlugins.alternatives</li>
    <li>beetsExternalPlugins.copyartifacts</li>
    <li>beetsExternalPlugins.extrafiles</li>
    <li>byzanz</li>
    <li>clementine</li>
    <li>denemo</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>39 packages built successfully:</summary>
  <ul>
    <li>ffmpeg-full (arcan.ffmpeg)</li>
    <li>brasero</li>
    <li>brasero-original</li>
    <li>corrscope</li>
    <li>cozy</li>
    <li>eolie</li>
    <li>gmrender-resurrect</li>
    <li>gnome.cheese</li>
    <li>gnome.totem</li>
    <li>gscrabble</li>
    <li>gst_all_1.gst-plugins-ugly</li>
    <li>handbrake</li>
    <li>imagination</li>
    <li>liba52</li>
    <li>lollypop</li>
    <li>mopidy-iris</li>
    <li>mopidy-mopify</li>
    <li>mopidy-mpd</li>
    <li>mopidy-mpris</li>
    <li>mopidy-musicbox-webclient</li>
    <li>mopidy-podcast</li>
    <li>mopidy-scrobbler</li>
    <li>mopidy-somafm</li>
    <li>mopidy-soundcloud</li>
    <li>mopidy-subidy</li>
    <li>mopidy-tunein</li>
    <li>mopidy-youtube</li>
    <li>mpd</li>
    <li>mpd-small</li>
    <li>pithos</li>
    <li>printrun</li>
    <li>python38Packages.moderngl-window</li>
    <li>python38Packages.pyglet</li>
    <li>python38Packages.pytmx</li>
    <li>python39Packages.moderngl-window</li>
    <li>python39Packages.pyglet</li>
    <li>python39Packages.pytmx</li>
    <li>qt-video-wlr</li>
    <li>rpiplay</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/liba52/default.nix:30:3:
```
   |
30 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/libraries/liba52/default.nix:21:5:
```
   |
21 |     ./no-always-inline.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/libraries/liba52/default.nix:20:5:
```
   |
20 |     ./no-disable-pic.patch
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

emilazy committed:
handbrake: remove liba52 dependency

This hasn't been used since 2014; see
https://github.com/HandBrake/HandBrake/commit/0306771b4a5c09d5bbd5742221eb0bc52103f2c5.

emilazy commented:
Force-pushed to (hopefully) fix VLC. See https://github.com/NixOS/nixpkgs/pull/137942/commits/47856660b42b95f14a3ead89e08fc241b1170ef5 for the annoying details.

risicle commented:
`handbrake` fails for me on macos 10.15:

```
builder for '/nix/store/bl50manhqg7nashcajkmkfdn4ifxrqqr-handbrake-1.4.1.drv' failed with exit code 2; last 10 log lines:
                     ^
  ../libhb/platform/macosx/vt_common.m:169:93: error: use of undeclared identifier 'kVTProfileLevel_HEVC_Main10_AutoLevel'; did you mean 'kVTProfileLevel_H264_Main_AutoLevel'?
              vt_h265_10bit_available = is_hardware_encoder_available(kCMVideoCodecType_HEVC, kVTProfileLevel_HEVC_Main10_AutoLevel);
                                                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                                                              kVTProfileLevel_H264_Main_AutoLevel
  /nix/store/avz81w2jaikr5hqs0q3n837h4ld67prf-apple-framework-VideoToolbox/Library/Frameworks/VideoToolbox.framework/Headers/VTCompressionProperties.h:258:29: note: 'kVTProfileLevel_H264_Main_AutoLevel' declared here
  VT_EXPORT const CFStringRef kVTProfileLevel_H264_Main_AutoLevel API_AVAILABLE(macosx(10.9), ios(8.0), tvos(10.2));
                              ^
  1 warning and 1 error generated.
  make: *** [../libhb/module.rules:15: libhb/platform/macosx/vt_common.o] Error 1
```

emilazy commented:
It looks like the VideoToolbox dependency was present before b725cd70c941c2d1754f48a3a4a75e1f6111b826, so I guess this is a case of the updated version depending on things that aren't in the old headers we use for `x86_64-darwin`. Maybe we could drop VideoToolbox support for non-Apple Silicon? That's not very satisfying though.

@risicle Could you check if it works with this (possibly evil, this doesn't seem to be done in-tree at present) change?

```patch
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index aca7715bc5a..ab9c66af554 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -25580,7 +25580,8 @@ with pkgs;
   lxdvdrip = callPackage ../applications/video/lxdvdrip { };
 
   handbrake = callPackage ../applications/video/handbrake {
-    inherit (darwin.apple_sdk.frameworks) AudioToolbox Foundation VideoToolbox;
+    inherit (darwin.apple_sdk.frameworks) AudioToolbox Foundation;
+    inherit (darwin.apple_sdk_11_0.frameworks) VideoToolbox;
     inherit (darwin) libobjc;
   };
 
```

risicle commented:
```
error: attribute 'apple_sdk_11_0' missing, at /Users/bob/.cache/nixpkgs-review/pr-137942/nixpkgs/pkgs/top-level/all-packages.nix:25764:14
```

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -16326,7 +16326,9 @@ with pkgs;
   icu = icu69;
 
-  id3lib = callPackage ../development/libraries/id3lib { };
+  id3lib = callPackage ../development/libraries/id3lib {
+    inherit (darwin) libiconv;
+  };
```
```suggestion
  id3lib = callPackage ../development/libraries/id3lib { };
```
libiconv is part of top-level.

SuperSandro2000 commented on pkgs/development/libraries/liba52/default.nix:
```diff
@@ -0,0 +1,36 @@
+    description = "ATSC A/52 stream decoder";
+    homepage = "https://code.videolan.org/videolan/liba52";
+    platforms = platforms.unix;
+    license = licenses.gpl2Plus;
```
Missing maintainer

SuperSandro2000 commented on pkgs/applications/video/handbrake/default.nix:
```diff
@@ -46,27 +48,18 @@
 
-  # Remove with a release after 1.3.3
-  patches = [
-    (fetchpatch {
```
Please remove the unused fetchpatch from inputs.

Sciencentistguy commented:
Any news on this PR moving forward?

marcelarie commented:
👀

Artturin commented:
@marcelarie @Sciencentistguy feel free to pick it up

SuperSandro2000 commented:
Closing due to inactivity
